debezium-connector-mongodb/src/test/java/io/debezium/connector/mongodb/transforms/ExtractNewDocumentStateTestIT.java

```diff
@@ -1387,6 +1388,33 @@ public void testAddFieldsHandleDeleteRewriteAndTombstone() throws Exception {
         assertThat(tombstoneTransformed.value()).isNull();
     }
 
+    @Test
+    @FixFor("DBZ-2585")
+    public void testEmptyArray() throws InterruptedException, IOException {
+        final Map<String, String> transformationConfig = new HashMap<>();
+        transformationConfig.put("array.encoding", "array");
+        transformationConfig.put("sanitize.value", "true");
+        transformation.configure(transformationConfig);
+
+        // Test insert
+        primary().execute("insert", client -> {
+            client.getDatabase(DB_NAME).getCollection(this.getCollectionName())
+                    .insertOne(Document.parse("{'empty_array': [] }"));
+        });
+        try {
+            SourceRecords records = consumeRecordsByTopic(1);
+            assertThat(records.recordsForTopic(this.topicName()).size()).isEqualTo(1);
+
+            final SourceRecord insertRecord = records.recordsForTopic(this.topicName()).get(0);
+            final SourceRecord transformedInsert = transformation.apply(insertRecord);
+            assertThat(transformedInsert.valueSchema().fields().stream().map(Field::name).filter(n -> n.equals("empty_array")).count()).isEqualTo(0);
```

Comment:
Couldn't you simply do `assertThat(transformedInsert.valueSchema().field("empty_array")).isNull()`?